makefile

```diff
@@ -291,6 +306,16 @@ else
 	@echo 'Info: Pylint requires Python 2; skipping this step on Python $(python_major_version)'
 endif
 
+flake8.log: makefile $(flake8_rc_file) $(flake8_py_files)
+	rm -f flake8.log
+	-bash -c "set -o pipefail; PYTHONPATH=. flake8 --config=$(flake8_rc_file) $(flake8_py_files) 2>&1 |tee flake8.tmp.log"
```

Comment:
I propose to make this last one a separate issue since it is currently finding a number of things. Lets get the function in then make changes to the checks, etc.  Note that today, we also see a lot of errors in flake8 that are disabled in pylint.  To disable these we will have to go back and do disable in the the code for lots of things.